lib/vrl/cli/src/repl.rs

```diff
@@ -158,21 +169,40 @@ fn resolve(
 struct Repl {
     highlighter: MatchingBracketHighlighter,
     validator: MatchingBracketValidator,
-    hinter: HistoryHinter,
+    history_hinter: HistoryHinter,
     colored_prompt: String,
+    hints: Vec<String>,
 }
 
 impl Repl {
     fn new() -> Self {
         Self {
             highlighter: MatchingBracketHighlighter::new(),
-            hinter: HistoryHinter {},
+            history_hinter: HistoryHinter {},
             colored_prompt: "$ ".to_owned(),
             validator: MatchingBracketValidator::new(),
+            hints: initial_hints(),
         }
     }
 }
 
+fn initial_hints() -> Vec<String> {
```

Comment:
As far as I can tell, both `f.identifier()` and `RESERVED_TERMS` return static strings, so you might as wel return `Vec<&'static str>` here.

Doesn't really matter too much given that the REPL is not (does not have to be) optimized for performance, but I thought I'd point it out nonetheless.

Reply:
It's never a bad time to remember that `String`s can be a crutch 🤣